_posts/2018-07-13-KafkaSchemaRegistryPart1.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title: Kafka - Confluent Schema Registry. Part 1
+categories: Kafka
+tags: [Kafka, Schema registry, Apache Avro, Confluent]
+author: NikitaZhevnitskiy
+---
+
+# Introduction
+
+Let's talk about Kafka Schema Registry essentials.
+
+# Repository
+
+Repository contains examples of schema evolution - full compatibility.
+[Kafka-schema-registry-poc](https://github.com/sysco-middleware/kafka-schema-registry-poc)
+
+## Content: 
+
+1. What is schema registry and why we need it?  
+1.1. Use case.  
+1.2. Full compatibility.  
+1.3. Avro serializer.
+2. How to define schema?  
+2.1. In source code.  
+2.2. Using reflection.  
+2.3. Using json.
+3. Schema evolution example.  
+3.1. Up kafka.  
+3.2. Up project.
+4. Important notes.
+
+## 1. What is schema registry and why we need it?
+
+Kafka schema registry is service(app) which stores schemas and provides RESTful interface to preform CRUD operations. 
```

Comment:
same as before. `s/Kafka shema registry/Confluent Schema Registry/g`

Reply:
Also, not sure what "service(app)" means. Maybe "application" is enough?
Then, strictly, it doesn't store schemas, but manage compatibility. Store schemas *in* Kafka.
Finally, not necessarially CRUD operations, it is a bit more than that. (e.g. check compatibility).

Reply:
```
The Schema Registry stores a versioned history of all schemas and allows for the evolution of schemas according to the configured compatibility settings and expanded Avro support
``` https://docs.confluent.io/current/schema-registry/docs/index.html 

and again `storing and retrieving` https://docs.confluent.io/current/schema-registry/docs/intro.html 

I agree that CRUD operations in not convenient here

Reply:
That's better. Just clarify that the actual storage happens on Kafka itself.